python/semantic_kernel/orchestration/sk_function.py

```diff
@@ -115,40 +122,88 @@ def from_semantic_config(
         if function_config is None:
             raise ValueError("Function configuration cannot be `None`")
 
-        async def _local_func(client, request_settings, context):
+        async def _local_func(client, request_settings, context: 'SKContext', **kwargs):
             if client is None:
                 raise ValueError("AI LLM service cannot be `None`")
 
             try:
-                if function_config.has_chat_prompt:
-                    as_chat_prompt = cast(
-                        ChatPromptTemplate, function_config.prompt_template
-                    )
+                if not function_config.has_chat_prompt:
+                    prompt = await function_config.prompt_template.render_async(context)
+                    completion = await client.complete_async(prompt, request_settings)
+                    context.variables.update(completion)
+                    return context
+            except Exception as e:
+                # TODO: "critical exceptions"
+                context.fail(str(e), e)
+                return context
 
-                    # Similar to non-chat, render prompt (which renders to a
-                    # list of <role, content> messages)
-                    messages = await as_chat_prompt.render_messages_async(context)
+            functions = kwargs.get("functions")
```

Comment:
This could be context.skill_collection after removing the `functions_calling_enabled`

Reply:
correct